### PR TITLE
status-notifier: Plug large leak with icons from paths

### LINKED ISSUE #1444: notification_area applet memory leaks if the tray icon is updated constently
#### Expected behaviour
No memory leaks.

#### Actual behaviour
mate-panel will memory leak really fast until no ram is left.
Video showcasing the problem: https://www.youtube.com/watch?v=zjMsdzgMdvE

#### Steps to reproduce the behaviour
1. Clone and compile [runcat-tray](https://github.com/chux0519/runcat-tray)
2. Open runcat-tray while using mate-panel with the notification area applet.

runcat-tray will call `app_indicator_set_icon` every tick to update the tray icon, this will result in mate-panel memory leaking.
I haven't tested with others apps and/or checked the notification area applet code yet.

#### MATE general version
MATE 1.28.2

#### Package version
mate.mate-panel 1.28.1

#### Linux Distribution
NixOS unstable 24.11.20240612.57d6973 (Vicuña) x86_64

#### Link to bugreport of your Distribution (requirement)
This issue doesn't seem to be distribution related, I will check on linux mint tomorrow and return results if necessary.

### PR DESCRIPTION
**Pull request by cwendling**
When loading an icon from a path rather than the icon theme, we scale it down to match the scale we need[^1], but we leaked the original pixbuf in this case.

Fixes #1444.

[^1]: See commit e5c8646d8f6167fe2d3634a4604a1bf9ec8ee5af

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
This builds and runs fine. Running an old test case than changes an indicator icon every  second for an unrelated bug, I did not see any rapid RAM use runup, though with other memory leaks anywhere on the system it's hard to be sure.

The code change however should definately work.

**Comment by lukefromdc** [comment]
Just rebased against master for the travis fix, will merge if that passes as we now have a second successful test and one which verifies this fixed the memory leak at
**https://github.com/mate-desktop/mate-panel/issues/1444#issuecomment-2176307508**

**Comment by lukefromdc** [comment]
Cannot update translations, the new tzm translation files still generate errors. Thus will release 1.28.2 with the existing translations. This has broken updating translations in every package

**Comment by lukefromdc** [comment]
cherrypicked to 1.28 and release 1.28.2 with existing translations due to build failure if translations are updated.

**Comment by lukefromdc** [comment]
Anyone have a problem with cherrypicking to 1.26 as well? Lots of distro versions shipping that won't be updating to 1.28. Please advise

**Comment by cwendling** [comment]
@lukefromdc fine with me, it's a big enough leak that it's worth backporting, and it should apply cleanly there as well.

**Comment by lukefromdc** [comment]
Will do, thanks

**Comment by lukefromdc** [comment]
On trying to update travis.yml  so a release would be possible I got this error:
`The command "./docker-build --name ${DISTRO} --config .build.yml --install" failed and exited with 1 during`
Will try reverting the travis change and cherrypicking to the 1.26 branch and see what happens.

**Comment by lukefromdc** [comment]
./autogen.sh  --with-in-process-applets=wncklet,clock,fish,notification-area --prefix=/usr libdir=/usr/lib/x86_64-linux-gnu --sysconfdir=/etc --enable-shared=yes --enable-static=no --disable-maintainer-mode --disable-schemas-compile

**Comment by lukefromdc** [comment]
Same travis error with or without the deploy fix. I do not know how to fix that, so until that is fixed I cannot do a release for 1.26 carrying this cherrypick

**Comment by raveit65** [comment]
@lukefromdc 
Try `dist: jammy`, see master/1.28 branch

**Comment by lukefromdc** [comment]
That solved the problem, will release 1.26.5 again using the existing translations due to breakage in seemingly all repos with the new ones

**Comment by lukefromdc** [comment]
Fixed, 1.26.1 (after the other travis fix, which I had to fiddle with) is out so all recent distros should pick this up

**Pull request merged**
